src/org/labkey/cds/CDSController.java

```diff
@@ -1370,4 +1373,30 @@ public void setReferrer(String referrer)
         }
 
     }
+
+    @RequiresPermission(ReadPermission.class)
+    public class GetReportsDataAction extends ReadOnlyApiAction<Object>
+    {
+        @Override
+        public Object execute(Object o, BindException errors) throws Exception
+        {
+            Map<String, JSONObject> result = new HashMap<>(); //key = rowId, val = reportName
+            Collection<Report> reports = ReportService.get().getReports(getUser(), getContainer());
+            if (reports.size() > 0)
+            {
+                for (Report report : reports)
+                {
+                    if (report.getDescriptor().isShared())
+                    {
+                        JSONObject obj = new JSONObject();
+                        obj.put("reportName", report.getDescriptor().getReportName());
+                        obj.put("url", report.getRunReportURL(getViewContext()));
```

Comment:
I'm not clear from the spec where those url links to. From the code it seems it's going to take user to a LabKey server report page, that's not CDS app?

Reply:
See usages in store/Study.js line 171
See usage in store/Publication.js line 113

Reply:
Right, but it looks like the link takes to LKS page, not within CDS page?

Reply:
oh yes, that is correct.

Reply:
that we might want to confirm with client. we had previously displayed R report in CDS app so that might be their expectation here as well

Reply:
Ah got it! We are meeting with them today, and will bring this up with them. Thank you, Xing!

Reply:
Yup, you are right! It needs to be displayed in the CDS app - ill get that fixed. 
Good catch, Xing!

Reply:
Now updated the url, it takes user to the report within the cds app.